modules/asset-registry/src/lib.rs

```diff
@@ -275,6 +280,88 @@ impl<T: Config> ForeignAssetIdMapping<ForeignAssetId, MultiLocation, AssetMetada
 	}
 }
 
+pub struct FixedRateOfForeignAsset<T, F: Get<u128>, R: TakeRevenue>(
+	Weight,
+	u128,
+	Option<MultiLocation>,
+	PhantomData<(T, F, R)>,
+);
+
+impl<T: Config, F: Get<u128>, R: TakeRevenue> WeightTrader for FixedRateOfForeignAsset<T, F, R> {
+	fn new() -> Self {
+		Self(0, 0, None, PhantomData)
+	}
+
+	fn buy_weight(&mut self, weight: Weight, payment: Assets) -> Result<Assets, XcmError> {
+		log::trace!(target: "asset-registry::weight", "buy_weight weight: {:?}, payment: {:?}", weight, payment);
+
+		for (index, multi_asset) in payment.fungible.iter().enumerate() {
```

Comment:
You can use `fungible.iter().next()` to get the first element.

Reply:
Will there be multiple Assets here? I want to support all, but for the time being only support the first one

Reply:
We can always update the impl if multiple assets support is needed in the future.